### ISSUE TITLE
Wrong hasher for of DefaultBinaryTree

### ISSUE THREAD

**Issue by vmx** [filecoin-project/rust-fil-proofs#1606]
### Description

The `DefaultBinaryTree` is defined as `BinaryMerkleTree<DefaultTreeHasher>`, while `DefaultTreeHasher = PoseidonHasher`: https://github.com/filecoin-project/rust-fil-proofs/blob/aecc1eeca5d6f187875114dc1a5f5f9abab96b69/filecoin-proofs/src/constants.rs#L144

Though the `DefaultBinaryTree` is also used at code paths which use SHA2 hashing, e.g.: https://github.com/filecoin-project/rust-fil-proofs/blob/aecc1eeca5d6f187875114dc1a5f5f9abab96b69/filecoin-proofs/src/api/seal.rs#L133.

The current code works, because the code paths the `DefaultBinaryTree` is used on, it only cares about the arity and the size of the hashes, which is the same for SHA2 and Poseidon. Though for correctness and clarity, the correct hasher for `DefaultBinaryTree` should be used.

### Acceptance criteria

The correct hashers are uses for the corresponding trees.

### Risks + pitfalls

Sometimes the Poseidon hasher might be needed. It needs thorough review whether changing the `DefaultBinaryTree` to SHA2 is enough, or whether some new type needs to be introduced.

### Where to begin

Look at all occurrences of `DefaultBinaryTree` and check if SHA2 or Poseidon hashing should be used.

**Comment by vmx** [comment]
I've assigned this issue to @storojs72 as you are familiar with the merkle trees and it's a good way to dig deeper into how they are used.

**Comment by storojs72** [comment]
Yes, I agree. We indeed have some confusing places in the codebase (where it is not obvious whether Poseidon or Sha256 hasher should be really used) that require clarification

**Comment by vmx** [comment]
As per https://github.com/filecoin-project/rust-fil-proofs/issues/1783  
 - Potential impact of this work: It would make the code easier to understand and reason about.
 - Rough dev effort estimate: 1 day
 - Dependencies: None
 - Next steps: Do it

**Comment by cryptonemo** [comment]
Not going to be resolved.  Re-open if needed.
